readthedocs/projects/tasks.py

```diff
@@ -382,8 +383,48 @@ def update_documentation_type(self):
         api_v2.project(self.project.pk).put(project_data)
         self.project.documentation_type = ret
 
+    def update_programming_language(self):
+        """Update a project's programming language based on linguist output"""
+        language = self.get_top_programming_language()
+        if language:
+            project_data = api_v2.project(self.project.pk).get()
+            if project_data['programming_language'] != language:
+                log.info(
+                    u'Updating primary programming lang from "{}" to "{}"'.format(
+                        project_data['programming_language'], language
+                    ))
+                project_data['programming_language'] = language
+                api_v2.project(self.project.pk).put(project_data)
+                self.project.programming_language = language
+
+    def get_top_programming_language(self):
+        """
+        Get the top programming language from Linguist output
+
+        Only retrieves the language if it is a known language to RTD. Linguist
+        knows of a lot more language than RTD.
+        """
+        language = None
+
+        language_reverse_mapping = dict((y, x) for x, y in PROGRAMMING_LANGUAGES)
+        artifact_path = self.project.artifact_path(
+            version=self.version.slug, type_='linguist')
+        linguist_json_path = os.path.join(artifact_path, 'linguist.json')
+
+        if os.path.isfile(linguist_json_path):
+            with open(linguist_json_path, 'r') as fd:
+                data = json.load(fd)
+                if data:
+                    _, lang = data[0]
+                    if lang in language_reverse_mapping:
+                        language = language_reverse_mapping[lang]
+                    else:
+                        log.warning(u'Unknown top programming language "{}"'.format(lang))
```

Comment:
Not sure we really want to warn here. This feels more like a `debug` statement. What can we do about the warning?

Reply:
My thought process here is that if we see enough of a programming language that is popular on RTD but not supported we could add support for it. Perhaps an `info` is better.